src/common/components/elements/AppHeader.tsx

```diff
@@ -209,6 +111,124 @@ export function AppHeader() {
   );
 }
 
+function DiscoverMenu(): JSX.Element {
+  const [isTouchAndShowItems, setIsTouchAndShowItems] = useState<boolean>(false);
+  const [isTouch, setIsTouch] = useState<boolean>(false);
+  const [forceHide, setForceHide] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsTouch(isTouchScreenOnly());
+  }, [setIsTouch]);
+
+  const onClickHeader = useCallback(() => {
+    if (isTouch) setIsTouchAndShowItems(!isTouchAndShowItems);
+  }, [isTouchAndShowItems, isTouch]);
+
+  const onClickHideTemporarily = useCallback(() => {
+    setIsTouchAndShowItems(false);
+    setForceHide(true);
+    setTimeout(() => setForceHide(false), 100);
+  }, []);
+
+  let containerDisplayClasses: string | undefined;
+  let buttonDisplayBasedTextClasses: string;
+  let arrowDownDisplayClasses: string;
+  let arrowUpDisplayClasses: string;
+  let itemContainerDisplayClasses: string;
+  if (!isTouch) {
+    containerDisplayClasses = 'group';
+    buttonDisplayBasedTextClasses = 'text-gray-300 hover:text-white focus:text-white group-hover:text-white group-focus:text-white';
+    arrowDownDisplayClasses = forceHide ? 'block' : 'block group-hover:hidden group-focus:hidden';
+    arrowUpDisplayClasses = forceHide ? 'hidden' : 'hidden group-hover:block group-focus:block';
+    itemContainerDisplayClasses = forceHide ? 'hidden' : 'hidden group-hover:block group-focus:block';
+  } else if (isTouchAndShowItems) {
+    buttonDisplayBasedTextClasses = 'text-white group-focus:text-white';
+    arrowDownDisplayClasses = forceHide ? 'block' : 'hidden group-focus:block';
+    arrowUpDisplayClasses = forceHide ? 'hidden' : 'block group-focus:hidden';
+    itemContainerDisplayClasses = forceHide ? 'hidden' : 'block group-focus:block';
+  } else {
+    buttonDisplayBasedTextClasses = 'text-gray-300 group-focus:text-white';
+    arrowDownDisplayClasses = 'block';
+    arrowUpDisplayClasses = 'hidden';
+    itemContainerDisplayClasses = forceHide ? 'hidden' : 'hidden group-focus:block';
+  }
```

Comment:
Oh see you already using it why not use the object notation bc of the nested conditions?

Reply:
I love the classNames utility. 

A couple of reasons for this approach here:
1. yes, nested conditions: IMO it will be messy and confusing to combine those in a single statement
2. This more clearly shows the styling over the whole set of components when in each state. Generally I think it's fine to describe the different states inline in the className, but with the number of elements being affected and there being several states, this feels more readable.